File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        
        int size = prices.size();
        vector<int>maxsum(size);
        
        int max = INT_MIN;
        for(int i = size-1; i>=0; i--){
            
            if(prices[i] > max)
            {
                max = prices[i];
                
            }
            maxsum[i] = max;
                
        }
        
        int ans =0;
        for(int i=0 ;i<size; i++){
            
            int cur = maxsum[i] - prices[i];
            if(cur > ans)
                ans = cur;
            
        }
        
        
        return ans;
        
        
        
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        
        maxi =0
        mini =prices[0]
        
        for i in range(len(prices)):
            maxi=  max(maxi,prices[i] - mini)
            mini = min(mini,prices[i])
            
        return maxi
            
            
        